musttag.go

```diff
@@ -149,31 +149,43 @@ func (c *checker) checkType(typ types.Type, tag string) bool {
 	}
 	c.seenTypes[typ.String()] = struct{}{}
 
-	if implementsInterface(typ, c.ifaceWhitelist, c.imports) {
-		return true // the type implements a Marshaler interface; see issue #64.
-	}
-
-	styp, ok := c.parseStruct(typ)
-	if !ok {
+	styp, shouldCheckStruct := c.unwrapStructType(typ)
+	if !shouldCheckStruct {
 		return true // not a struct.
 	}
 
 	return c.checkStruct(styp, tag)
 }
 
-func (c *checker) parseStruct(typ types.Type) (*types.Struct, bool) {
+// recursively unpack a type to the next type that needs checking,
+// until we get to some underlying struct for checking
+//
+//	SomeStruct -> struct{SomeStructField: ... }
+//	[]*SomeStruct -> struct{SomeStructField: ... }
+//	...
+//
+// exits early if it hits a type that implements a whitelisted interface
+func (c *checker) unwrapStructType(typ types.Type) (*types.Struct, bool) {
```

Comment:
```suggestion
func (c *checker) parseStruct(typ types.Type) (*types.Struct, bool) {
```

IMHO, the original naming is better.